src/oracle_implementations/discount_rate/NotionalFinance/NotionalFinanceValueProvider.sol

```diff
@@ -112,4 +105,31 @@ contract NotionalFinanceValueProvider is Oracle, Convert {
         // The result is a 59.18 fixed-point number.
         return discreteRateD59x18;
     }
+
+    /// @notice Retrieve the oracle rate from the NotionalFinance Market
+    function getOracleRate() internal view returns (uint256) {
+        uint256 settlementDate = getSettlementDate();
+        // Attempt to retrieve the oracle rate directly by using the maturity and settlement date
+        MarketParameters memory marketParams = INotionalView(notionalView)
+            .getMarket(uint16(currencyId), maturityDate, settlementDate);
+
+        // If we find an active market we can return the oracle rate otherwise we revert
+        if (marketParams.oracleRate <= 0) {
+            revert NotionalFinanceValueProvider__getValue_invalidMarketParameters(
+                currencyId,
+                maturityDate,
+                settlementDate
+            );
+        }
+
+        return marketParams.oracleRate;
+    }
+
+    /// @notice Computes the settlement date as is done in the NotionalFinance contracts
+    /// Reference 1: https://github.com/notional-finance/contracts-v2/blob/master/contracts/internal/markets/DateTime.sol#L14
+    /// Reference 2:
+    /// https://github.com/notional-finance/contracts-v2/blob/d89be9474e181b322480830501728ea625e853d0/contracts/internal/markets/Market.sol#L536
+    function getSettlementDate() public view returns (uint256) {
+        return block.timestamp - (block.timestamp % QUARTER) + QUARTER;
```

Comment:
It feels much cleaner simplifying this to

`block.timestamp / QUARTER + QUARTER`

Reply:
wait but this is not the same. say timestamp = 19 and quarter = 5

v1: 19 - ( 4) + 5 = 20
v2: 19 / 5 + 5 = 8

@cleanunicorn

Reply:
You're absolutely right. I was thinking of 

`block.timestamp / QUARTER * QUARTER + QUARTER`

But doesn't look too clean.

Reply:
I would keep it as it is because it's very easy to see that it's the same calculation as in the Notional contracts.